mlflow/store/artifact/utils/models.py

```diff
@@ -63,28 +57,31 @@ def _parse_model_uri(uri):
     if parsed.scheme != "models":
         raise MlflowException(_improper_model_uri_msg(uri))
     path = parsed.path
-    m = _MODEL_URI_REGEX.match(path)
-    if m is None:
+    if not path.startswith("/") or len(path) <= 1:
         raise MlflowException(_improper_model_uri_msg(uri))
-    gd = m.groupdict()
-    model_name = gd.get("model_name")
-    suffix = gd.get("suffix")
-    alias = gd.get("alias")
-    if (model_name.strip() == "") or (suffix and alias) or (suffix is None and alias is None):
+
+    parts = path[1:].split("/")
```

Comment:
for a uri like `models://////foo`, how many slashes does `path` have? Is it always sufficient to do `path[1:]`?

Reply:
Yeah so that part I reverted from prior to https://github.com/mlflow/mlflow/pull/8164. How I understand it is that the `parser` takes care of the additional slashes and `parsed.path` only leaves us with 1, such as with this test case: 
```
("models://scope:key/Ads Model 1@None", "Ads Model 1", "None"),
```
That being said, I can add a test case to confirm that something like `models://////foo` would pass

Reply:
Ah actually, upon adding that test case, it seems that currently we fail on a URI like `models:///////Ads Model 1/12345`. However, this was the logic that we had before as well. Do we want to change this going forward?

Reply:
Would be nice to fix :D

Reply:
I was reading into the [current parser](https://docs.python.org/3/library/urllib.parse.html) we use, and technically would something like `models://////foo` be a valid path? Or do we just want to add custom logic to support such URIs going forward?

Reply:
Excellent point to think about.... don't Windows Network drives use something like this (or is it 4? I can't remember... I know it's something wacky)

Reply:
Looks like currently, `models://foo` doesn't even work, so I guess this has been broken all this time 😬

Reply:
Added code to use `lstrip` but let me know if that's actually what we want